lib/logging.dart

```diff
@@ -61,6 +61,10 @@ class Logger {
     return _loggers.putIfAbsent(name, () => new Logger._named(name));
   }
 
+  factory Logger.orphan(String name, {Logger parent, Map<String, Logger> children: const {}}) {
```

Comment:
If we go with this approach, we might want to rename this to `Logger.detached`?

For the dartdoc: it would be great to highlight that each `new Logger.detached` returns a new instance (unlike `new Logger`) and that the logger is not attached to the hierarchy of loggers that you normally create with the default constructor.

Also, it would be great to add a couple tests to illustrate that two detached loggers with the same name are different to each other and to a logger created with new Logger.